src/styles/theme-toggle.css

```diff
@@ -1,280 +1,155 @@
-   Text labels (visually hidden but accessible for screen readers)
-   ------------------------------------------------------------------------- */
-.theme-toggle__label {
-  position: absolute;
-  width: 1px;
-  height: 1px;
-  padding: 0;
-  margin: -1px;
-  overflow: hidden;
-  clip: rect(0, 0, 0, 0);
-  white-space: nowrap;
-  border: 0;
-  border: 1px solid red; /* TODO: Remove debug border */
-}
-
-/* -------------------------------------------------------------------------
-   Icon containers (left = day/sun, right = night/moon)
-   ------------------------------------------------------------------------- */
-.theme-toggle__side {
-  width: 50%;
-  display: inline-flex;
-  align-items: center;
-  justify-content: center;
-  z-index: 1;
-  pointer-events: none;
-}
-
-/* -------------------------------------------------------------------------
-   Sun icon (left side) - Visible in light mode (unchecked)
-   ------------------------------------------------------------------------- */
-.theme-toggle__side--day::after {
-  content: "";
-  width: 1.4rem;
-  height: 1.4rem;
-  background-color: rgba(255, 255, 255, 0.95);
-  opacity: 0.95; /* Visible by default in light mode */
-  transition: opacity 0.2s ease;
-  -webkit-mask: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cg fill='none' stroke='%23fff' stroke-width='2' stroke-linecap='round'%3E%3Ccircle cx='12' cy='12' r='4'/%3E%3Cpath d='M12 2v2M12 20v2M4.93 4.93l1.41 1.41M17.66 17.66l1.41 1.41M2 12h2M20 12h2M4.93 19.07l1.41-1.41M17.66 6.34l1.41-1.41'/%3E%3C/g%3E%3C/svg%3E") no-repeat center / contain;
-  mask: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cg fill='none' stroke='%23fff' stroke-width='2' stroke-linecap='round'%3E%3Ccircle cx='12' cy='12' r='4'/%3E%3Cpath d='M12 2v2M12 20v2M4.93 4.93l1.41 1.41M17.66 17.66l1.41 1.41M2 12h2M20 12h2M4.93 19.07l1.41-1.41M17.66 6.34l1.41-1.41'/%3E%3C/g%3E%3C/svg%3E") no-repeat center / contain;
+    --toggle-track-border: rgba(255, 255, 255, 0.35);
+    --toggle-knob-size: 2.15rem;
+    --toggle-knob-offset: 0.35rem;
+    --theme-toggle-knob-outline-color: rgba(0, 0, 0, 0.55);
+    --sun-opacity: .95;
+    --moon-opacity: 0;
+    --logo-transition: translateX(-0.7rem);
+
+    margin-top: -0.25rem;
+
+    & label {
+        width: 9.26rem;
+        height: 2.73rem;
+        padding: 0.2rem;
+        margin-left: 0.2rem;
+        padding-right: 0rem;
+        position: relative;
+        display: block;
+        border-radius: 200px;
+        cursor: pointer;
+        background: var(--toggle-track-bg);
+        border: 1px solid var(--toggle-track-border);
+
+        &:hover {
+            background-color: rgba(255, 255, 255, 0.12);
+            transform: translateY(-0.125rem);
+            box-shadow: 0 0.5rem 1.5rem rgba(0, 0, 0, 0.15);
+        }
+
+        /* DARK MODE STYLES GO HERE */
+        &:has(input:checked) {
+            --toggle-track-bg:
```

Comment:
you really only need this one place to put the things that will change in dark mode. the selectors get really simplified if you do this.